ui/app/store/actions.js

```diff
@@ -304,86 +304,79 @@ export function importNewAccount (strategy, args) {
 
 export function addNewAccount () {
   log.debug(`background.addNewAccount`)
-  return (dispatch, getState) => {
+  return async (dispatch, getState) => {
     const oldIdentities = getState().metamask.identities
     dispatch(showLoadingIndication())
-    return new Promise((resolve, reject) => {
-      background.addNewAccount((err, { identities: newIdentities }) => {
-        if (err) {
-          dispatch(displayWarning(err.message))
-          return reject(err)
-        }
-        const newAccountAddress = Object.keys(newIdentities).find((address) => !oldIdentities[address])
-
-        dispatch(hideLoadingIndication())
 
-        forceUpdateMetamaskState(dispatch)
-        return resolve(newAccountAddress)
-      })
-    })
+    let newIdentities
+    try {
+      const { identities } = await promisifiedBackground.addNewAccount()
+      newIdentities = identities
+    } catch (error) {
+      dispatch(displayWarning(error.message))
+      throw error
+    }
+    const newAccountAddress = Object.keys(newIdentities).find((address) => !oldIdentities[address])
+    dispatch(hideLoadingIndication())
+    forceUpdateMetamaskState(dispatch)
+    return newAccountAddress
```

Comment:
Can we move these lines into the `try` block?

Reply:
That would change the behavior; the `catch` block would then be executed if any of these threw an exception, whereas beforehand it would not.

My goal with this PR was to have zero changes in behavior; just a pure refactor.

Reply:
Fair enough